tests/integration/targets/azure_rm_hdinsightcluster/tasks/main.yml

```diff
@@ -0,0 +1,223 @@
+- name: Prepare random number
+  set_fact:
+    rpfx: "{{ resource_group | hash('md5') | truncate(7, True, '') }}{{ 1000 | random }}"
+  run_once: yes
+
+- name: Create storage account
+  azure_rm_storageaccount:
+      resource_group: "{{ resource_group }}"
+      name: "storage{{ rpfx }}"
+      account_type: Standard_LRS
+      location: eastus2
+
+- name: Sample for Azure REST API - StorageAccounts_ListKeys
+  azure_rm_resource:
+    api_version: '2018-07-01'
+    method: POST
+    resource_group: "{{ resource_group }}"
+    provider: storage
+    resource_type: storageaccounts
+    resource_name: "storage{{ rpfx }}"
+    subresource:
+      - type: listkeys
+  register: storage_output
+
+- debug:
+    var: storage_output
+
+- name: Create instance of Cluster -- check mode
+  azure_rm_hdinsightcluster:
+    resource_group: "{{ resource_group }}"
+    name: "cluster{{ rpfx }}"
```

Comment:
why not make name a vriable as "cluster{{ rpfx }}"?

Reply:
i can change it upstream, and then it will be merged automatically to preview modules.
now i just want to have complete test here in sync with upstream